### PR TITLE
[SYCL][USM] Fix tests to deal with multiple CLs being installed

### PR DESCRIPTION
**Pull request by jbrodman**
Signed-off-by: James Brodman <james.brodman@intel.com>

### CODE REVIEW

File: sycl/test/usm/clext.cpp [hunk 1]
```diff
@@ -14,6 +14,37 @@
...
+bool findPlatformAndDevice(cl_device_type deviceType,
+			   cl_platform_id& platformOut,
+			   cl_device_id& deviceOut) {
+  cl_platform_id platforms[SIZE];
+  cl_uint numPlatforms;
+  cl_device_id device;
+  cl_int errorCode;
+  bool foundDevice = false;
+
+  errorCode = clGetPlatformIDs(SIZE, platforms, &numPlatforms);
```
Comment (bader) [inline sycl/test/usm/clext.cpp]: Here is the idiom to get the list of available platforms:
```c++
cl_uint numPlatforms;
cl_int errorCode = clGetPlatformIDs(0, nullptr, &numPlatforms);
std::vector<cl_platform_id> platforms(numPlatforms);
errorCode = clGetPlatformIDs(numPlatforms, platforms, nullptr);
```
This is common pattern to get a list of things from OpenCL API.

Also it might worth moving these common test infrastructure to the header file to not duplicate this code in other tests.

File: sycl/test/usm/clext.cpp [hunk 2]
```diff
@@ -14,6 +14,37 @@
...
 using namespace cl::sycl::detail::usm;
 
+static int SIZE = 6;
+
+bool findPlatformAndDevice(cl_device_type deviceType,
+			   cl_platform_id& platformOut,
+			   cl_device_id& deviceOut) {
+  cl_platform_id platforms[SIZE];
+  cl_uint numPlatforms;
+  cl_device_id device;
```
Comment (bader) [inline sycl/test/usm/clext.cpp]: Let's move `device` declaration to the line 32.

File: sycl/test/usm/clext.cpp (hunk 1 above)
Comment (jbrodman) [reply sycl/test/usm/clext.cpp]: This should be:

```
errorCode = clGetPlatformIDs(numPlatforms, platforms.data(), nullptr);
```

right?

File: sycl/test/usm/findplatforms.hpp [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+bool findPlatformAndDevice(cl_device_type deviceType,
+                           cl_platform_id &platformOut, cl_device_id &deviceOut,
+                           cl_int &errorCode) {
+  cl_uint numPlatforms;
+  bool foundDevice = false;
+
+  errorCode = clGetPlatformIDs(0, nullptr, &numPlatforms);
```
Comment (bader) [inline sycl/test/usm/findplatforms.hpp]: Can we just return `false` if any of OpenCL API calls return value != `CL_SUCCESS`?
This function overrides `errorCode` value multiple time w/o checking intermediate results. Probably we can avoid using `errorCode` variable at all.

Comment (jbrodman) [reply sycl/test/usm/findplatforms.hpp]: Yes.

Comment (bader) [reply sycl/test/usm/findplatforms.hpp]: How about checking `clGetPlatformIDs` error code?

File: sycl/test/usm/findplatforms.hpp [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+      std::vector<cl_device_id> devices(numDevices);
+      errorCode = clGetDeviceIDs(platform, deviceType, numDevices,
+                                 devices.data(), nullptr);
+      if (errorCode != CL_SUCCESS) return false;
+
+      if (numDevices) {
+        platformOut = platform;
+        deviceOut = devices[0];
+        foundDevice = true;
```
Comment (bader) [inline sycl/test/usm/findplatforms.hpp]: `foundDevice` variable seems to be also unnecessary as we are looking for one device only.
We can make code simpler if we just add `return true` here and replace all other instances of `foundDevice` with `false`.

Comment (jbrodman) [reply sycl/test/usm/findplatforms.hpp]: Yep.
